### PR TITLE
Linux Parsing Rule add support for ISO 8601 compatible like timestamps

### PR DESCRIPTION
**Pull request by cweltPA**
## Status
- [ ] In Progress
- [x] Ready
- [ ] In Hold - (Reason for hold)

## Related Issues
fixes: [XSUP-42968](https://jira-dc.paloaltonetworks.com/browse/XSUP-42968) 

## Description
Up until this point, the implemented parsing rule supported only [RFC 3164](https://datatracker.ietf.org/doc/html/rfc3164) compatible timestamps, which lack an explicit timezone and timestamp originating year. such as _`Oct  9 22:10:46`_.

This PR adds support for ISO 8601 style timestamps as well, which do contain an explicit timezone and year. Samples of new timestamps format that would be supported beginning from version 1.0.10: 
   - *`2024-10-28T14:30:55Z`*
   - *`2024-10-28T14:30:55-0300`*
   - *`2024-10-28T14:30:55.123+02:00`*  

## Must have
- [ ] Tests
- [ ] Documentation

### CODE REVIEW

**Review by OBavly — APPROVED** [review]

File: Packs/LinuxEventsCollection/README.md [hunk 1]
```diff
@@ -1,52 +1,75 @@
 # Linux Events Collection
-This pack includes XSIAM content.
 
-## Collect Events from Vendor
+## Collector Configuration
 
-In order to use the collector, you can use one of the following options to collect events from the vendor:
+You could use any one of the following options to collect Linux events into Cortex XSIAM:
```
Comment (julieschwartz18) [inline Packs/LinuxEventsCollection/README.md:5]: Recommend:
Use one of the following options to collect Linux events into Cortex XSIAM:

File: Packs/LinuxEventsCollection/README.md [hunk 2]
```diff
@@ -1,52 +1,75 @@
...
+  - [Broker VM Syslog Applet](#broker-vm)
+  - [Filebeat XDRC (XDR Collector)](#xdrc-xdr-collector)
 
-- [Linux Events Collection](#linux-events-collection)
-  - [Collect Events from Vendor](#collect-events-from-vendor)
-    - [Broker VM](#broker-vm)
-    - [XDRC (XDR Collector)](#xdrc-xdr-collector)
-
-In either option, you will need to configure the vendor and product for this specific collector.
+After the Cortex XSIAM Collector starts ingesting logs from the configured paths on the Linux servers, you could query the collected logs under the *`linux_linux_raw`* dataset.
```
Comment (julieschwartz18) [inline Packs/LinuxEventsCollection/README.md:9]: Recommend:
After the Cortex XSIAM collector starts ingesting logs from the Linux servers, you can query the logs under the *`linux_linux_raw`* dataset.

File: Packs/LinuxEventsCollection/README.md [hunk 3]
```diff
@@ -1,52 +1,75 @@
...
-You will need to use the information described [here](https://docs-cortex.paloaltonetworks.com/r/Cortex-XDR/Cortex-XDR-Pro-Administrator-Guide/Configure-the-Broker-VM).\
-You can configure the specific vendor and product for this instance.
+#### Prerequisites
 
-1. Navigate to **Settings** -> **Configuration** -> **Data Broker** -> **Broker VMs**.
-2. Right-click, and select **Syslog Collector** -> **Configure**.
-3. When configuring the Syslog Collector, set:
-   - vendor as linux
-   - product as linux
+You would need to have a Broker VM set and configured as described [here](https://docs-cortex.paloaltonetworks.com/r/Cortex-XSIAM/Cortex-XSIAM-Documentation/Broker-VM).
```
Comment (julieschwartz18) [inline Packs/LinuxEventsCollection/README.md:15]: Recommend:
You need to set and configure a Broker VM. For more information, see [Broker VM](https://docs-cortex.paloaltonetworks.com/r/Cortex-XSIAM/Cortex-XSIAM-Documentation/Broker-VM).

File: Packs/LinuxEventsCollection/README.md [hunk 4]
```diff
@@ -1,52 +1,75 @@
...
-                 `sudo timedatectl set-timezone UTC`
+#### Configuration Steps
 
+1. Navigate to **Settings** &rarr; **Configuration** &rarr; **Data Broker** &rarr; **Broker VMs**. 
+2. Go to the **APPS** column under the **Brokers** tab and add the **Syslog** app for the relevant broker instance. If the Syslog app already exists, hover over it and click **Configure**.
+3. Click **Add New**.
+4. When configuring the Syslog Collector, set the following parameters:
+   | Parameter     | Value    
+   | :---          | :---                    
+   | `Vendor`      | Enter **linux**.
```
Comment (julieschwartz18) [inline Packs/LinuxEventsCollection/README.md:25]: Remove the word "Enter" in the Value column

File: Packs/LinuxEventsCollection/README.md [hunk 5]
```diff
@@ -1,52 +1,75 @@
...
+#### Configuration Steps
 
+1. Navigate to **Settings** &rarr; **Configuration** &rarr; **Data Broker** &rarr; **Broker VMs**. 
+2. Go to the **APPS** column under the **Brokers** tab and add the **Syslog** app for the relevant broker instance. If the Syslog app already exists, hover over it and click **Configure**.
+3. Click **Add New**.
+4. When configuring the Syslog Collector, set the following parameters:
+   | Parameter     | Value    
+   | :---          | :---                    
+   | `Vendor`      | Enter **linux**. 
+   | `Product`     | Enter **linux**.
```
Comment (julieschwartz18) [inline Packs/LinuxEventsCollection/README.md:26]: Remove the word "Enter" in the value column

File: Packs/LinuxEventsCollection/README.md [hunk 6]
```diff
@@ -1,52 +1,75 @@
...
+                    fields:
+                      vendor: linux
+                      product: linux
+         ```
+4. Apply the configured Filebeat profile to the requested target Linux servers by attaching it to a policy as described [here](https://docs-cortex.paloaltonetworks.com/r/Cortex-XSIAM/Cortex-XSIAM-Documentation/Apply-profiles-to-collection-machine-policies).
+
+
+#### Supported Timestamp formats 
+##### Broker VM Syslog
+- [RFC 3164](https://datatracker.ietf.org/doc/html/rfc3164#section-4.1.2) compatible timestamps, in UTC, for example: *`Oct  8 19:44:40`*. Note: You can run the following command on the relevant Linux servers for configuring their timezone to UTC:
```
Comment (julieschwartz18) [inline Packs/LinuxEventsCollection/README.md:59]: Recommend:
new line for the note as follows
**Note:**
You can run the following command on the relevant Linux servers to configure their timezone to UTC:

File: Packs/LinuxEventsCollection/README.md [hunk 7]
```diff
@@ -1,52 +1,75 @@
...
+  sudo timedatectl set-timezone UTC
+  ```
+- [RFC 3339](https://www.rfc-editor.org/rfc/rfc3339) compatible like timestamps. Examples: 
+   - *`2024-10-28T14:30:55Z`*
+   - *`2024-10-28T14:30:55-0300`*
+   - *`2024-10-28T14:30:55.123+02:00`*
+  
+
+##### Filebeat XDR Collector
+- [RFC 3339](https://www.rfc-editor.org/rfc/rfc3339) compatible like timestamps. Examples:
```
Comment (julieschwartz18) [inline Packs/LinuxEventsCollection/README.md:70]: Recommend:
new line for the Examples as follows
**Examples:**
 - *`2024-10-28T14:30:55Z`*
 - *`2024-10-28T14:30:55-0300`*
 - *`2024-10-28T14:30:55.123+02:00`*

File: Packs/LinuxEventsCollection/README.md [hunk 8]
```diff
@@ -1,52 +1,75 @@
...
+   - *`2024-10-28T14:30:55.123+02:00`*
+  
+
+##### Filebeat XDR Collector
+- [RFC 3339](https://www.rfc-editor.org/rfc/rfc3339) compatible like timestamps. Examples: 
+   - *`2024-10-28T14:30:55Z`*
+   - *`2024-10-28T14:30:55-0300`*
+   - *`2024-10-28T14:30:55.123+02:00`*  
+
+**Note**: Events logs collected via Filebeat with timestamp formats other than those listed above would be assigned with the agent collection time.
```
Comment (julieschwartz18) [inline Packs/LinuxEventsCollection/README.md:75]: Recommend:
new line for note as follows
**Note:**
Event logs collected via Filebeat with timestamp formats other than those listed above are assigned the agent collection time.

File: Packs/LinuxEventsCollection/ReleaseNotes/1_0_10.md [hunk 9]
```diff
@@ -0,0 +1,7 @@
+
+#### Parsing Rules
+
+##### Linux Events Collection Parsing Rule
+
+- Improved implementation of the existing parsing rule. 
+- Added support for [RFC 3339](https://datatracker.ietf.org/doc/html/rfc3339) compatible like timestamps, such as *`2024-10-28T14:30:55+0300`*, *`2024-10-28T14:30:55.123-03:00`* and *`2024-10-28T14:30:55.123Z`*, on top of the already supported [RFC 3164](https://datatracker.ietf.org/doc/html/rfc3164) compatible timestamps.
```
Comment (julieschwartz18) [inline Packs/LinuxEventsCollection/ReleaseNotes/1_0_10.md:7]: Recommend:
Added support for [RFC 3339](https://datatracker.ietf.org/doc/html/rfc3339) compatible timestamps, such as *`2024-10-28T14:30:55+0300`*, *`2024-10-28T14:30:55.123-03:00`* and *`2024-10-28T14:30:55.123Z`*, in addition to the already supported [RFC 3164](https://datatracker.ietf.org/doc/html/rfc3164) compatible timestamps.

**Review by julieschwartz18 — COMMENTED** [review]
TW review done

**Comment by julieschwartz18** [comment]
Added docs-approved label. Please review the TW edits

File: Packs/LinuxEventsCollection/README.md (hunk 1 above)
Comment (cweltPA) [reply Packs/LinuxEventsCollection/README.md]: fixed: 
https://github.com/demisto/content/pull/36882/files#diff-547b4347d8e4a399e431acab3ca503579d3d23c0f8854847ee85c97343769646R5

File: Packs/LinuxEventsCollection/README.md (hunk 3 above)
Comment (cweltPA) [reply Packs/LinuxEventsCollection/README.md]: Fixed: 
https://github.com/demisto/content/pull/36882/commits/be339141a34bd645960ec4e017b1ab64e68a7424#diff-547b4347d8e4a399e431acab3ca503579d3d23c0f8854847ee85c97343769646R15

File: Packs/LinuxEventsCollection/README.md (hunk 5 above)
Comment (cweltPA) [reply Packs/LinuxEventsCollection/README.md]: Fixed:
https://github.com/demisto/content/pull/36882/commits/be339141a34bd645960ec4e017b1ab64e68a7424#diff-547b4347d8e4a399e431acab3ca503579d3d23c0f8854847ee85c97343769646R25

File: Packs/LinuxEventsCollection/README.md (hunk 4 above)
Comment (cweltPA) [reply Packs/LinuxEventsCollection/README.md]: Fixed

File: Packs/LinuxEventsCollection/README.md (hunk 6 above)
Comment (cweltPA) [reply Packs/LinuxEventsCollection/README.md]: Fixed: 
https://github.com/demisto/content/pull/36882/commits/be339141a34bd645960ec4e017b1ab64e68a7424#diff-547b4347d8e4a399e431acab3ca503579d3d23c0f8854847ee85c97343769646R61

File: Packs/LinuxEventsCollection/README.md (hunk 2 above)
Comment (cweltPA) [reply Packs/LinuxEventsCollection/README.md]: Fixed:
https://github.com/demisto/content/pull/36882/commits/be339141a34bd645960ec4e017b1ab64e68a7424#diff-547b4347d8e4a399e431acab3ca503579d3d23c0f8854847ee85c97343769646R9

File: Packs/LinuxEventsCollection/README.md (hunk 7 above)
Comment (cweltPA) [reply Packs/LinuxEventsCollection/README.md:73]: Fixed:
https://github.com/demisto/content/pull/36882/commits/be339141a34bd645960ec4e017b1ab64e68a7424#diff-547b4347d8e4a399e431acab3ca503579d3d23c0f8854847ee85c97343769646R78

File: Packs/LinuxEventsCollection/README.md (hunk 8 above)
Comment (cweltPA) [reply Packs/LinuxEventsCollection/README.md]: Fixed:
https://github.com/demisto/content/pull/36882/commits/be339141a34bd645960ec4e017b1ab64e68a7424#diff-547b4347d8e4a399e431acab3ca503579d3d23c0f8854847ee85c97343769646R79

File: Packs/LinuxEventsCollection/ReleaseNotes/1_0_10.md (hunk 9 above)
Comment (cweltPA) [reply Packs/LinuxEventsCollection/ReleaseNotes/1_0_10.md]: Fixed

**Pull request merged**